fiftyone/utils/cvat.py

```diff
@@ -3939,6 +3947,48 @@ def _upload_annotations(
             num_uploaded_tags = len(anno_resp["tags"])
             num_uploaded_tracks = len(anno_resp["tracks"])
 
+        server_id_map = self._map_cvat_server_id(anno_resp, attr_id_map)
+
+        return server_id_map
+
+    def _map_cvat_server_id(self, anno_resp, attr_id_map):
+        # Create mapping from CVAT server ID to FiftyOne ObjectId for each
+        # label
+        server_id_map = {}
+        dup_ids = []
+        types = ["shapes", "tags", "tracks"]
+        attr_id_map_rev = {}
+        for class_id, class_attr_map in attr_id_map.items():
+            class_attr_map_rev = {i: n for n, i in class_attr_map.items()}
+            attr_id_map_rev[class_id] = class_attr_map_rev
+
+        for anno_type, anno_list in anno_resp.items():
+            if anno_type not in types:
+                continue
+
+            for anno in anno_list:
+                server_id = anno["id"]
+                class_id = anno["label_id"]
+                for attr in anno["attributes"]:
+                    spec_id = attr["spec_id"]
+                    value = attr["value"]
+                    attr_name = attr_id_map_rev[class_id][spec_id]
+                    if attr_name == "label_id":
+                        if server_id in server_id_map:
```

Comment:
You don't need to worry about duplicate server IDs (if they change we can assume they will be new, not reused)

Reply:
I need to check if server IDs are shared between `tracks`, `shapes`, and `tags`. If not, then there could be duplicates. If they are shared, I'll remove this.

Reply:
If they aren't globally unique then you'd have to store nested server ID maps for each shape type, right?

Reply:
Right. Fixed. They aren't globally unique, every newly uploaded/annotated label is assigned an integer ID server-side that is the previously assigned integer ID incremented by 1 . Each group (`tags`, `tracks`, and `shapes`) store their own ID, so assuming every server starts with ID 0 or 1 for each group, there can be overlapping server IDs for a tag, track, and shape.